src/lib.rs

```diff
@@ -1040,6 +1050,20 @@ const fn is_keyof_u16(key: &[u8]) -> bool {
     matches!(key, b"tcp" | b"tcp6" | b"udp" | b"udp6")
 }
 
+fn is_valid_ipv4(key: &[u8], value: &[u8]) -> bool {
+    if key == b"ip" && value[0] - 0x80 == 0x04 {
+        return true;
+    }
+    false
+}
+
+fn is_valid_secp256k1(key: &[u8], value: &[u8]) -> bool {
+    if key == b"secp256k1" && value[0] - 0x80 == 0x21 {
```

Comment:
Here is applies the same as well, two potential panics plus I'm not sure what the rationale is for this check